//
// Copyright 2019 The Project Oak Authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

syntax = "proto3";

package oak.examples.private_set_intersection;

import "google/protobuf/empty.proto";

message SubmitSetRequest {
  string set_id = 1;
  repeated string values = 2;
}

message GetIntersectionRequest {
  string set_id = 1;
}

message GetIntersectionResponse {
  repeated string values = 1;
}

service PrivateSetIntersection {
  // Submit a private set that will be contributed to the set intersection specified by the set ID.
  // The number of submissions is limited for individual intersection.
  rpc SubmitSet(SubmitSetRequest) returns (google.protobuf.Empty);
  // Request the final private set intersection specified by the set ID.
  // Once the intersection is requested it becomes locked, and new contributions are discarded.
  rpc GetIntersection(GetIntersectionRequest) returns (GetIntersectionResponse);
}
